File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int num = 0;
    public int solution(int[] numbers, int target) {
        int answer = 0;
        dfs(numbers, 0, 0, target);
        return num;
    }
    
    public void dfs(int[] numbers, int sum, int idx, int target) {
        if (numbers.length == idx) {
            if (sum == target) {
                num++;
            }
            return;
        }
    
        dfs(numbers, sum + numbers[idx], idx + 1, target);
        dfs(numbers, sum - numbers[idx], idx + 1, target); 
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
var count = 0;
function solution(numbers, target) {
    dfs(numbers, target, 0, 0);
    return count;
}

function dfs (numbers, target, sum, idx) {
    if (idx == numbers.length) {
        if (target == sum) {
            count++;
        }
        return;
    }
    
    dfs(numbers, target, sum + numbers[idx], idx + 1);
    dfs(numbers, target, sum - numbers[idx], idx + 1);
}